### PR TITLE
feat(translate-mode): translate shapes in screen space

### PR DESCRIPTION
**Pull request by igorDykhta**
- an option to translate shapes in screen space, as direct translation in lon/lat space distorts shapes

image

### CODE REVIEW

File: modules/edit-modes/src/utils.ts [hunk 1]
```diff
@@ -457,3 +457,22 @@ function getEditHandlesForCoordinates(
   }
   return editHandles;
 }
+
+export function isNumeric(val: any) {
+  return !Array.isArray(val) && !isNaN(parseFloat(val)) && isFinite(val);
+}
+
+export function traverseCoords(
+  coords: any,
```
Comment (ibgreen) [inline modules/edit-modes/src/utils.ts:466]: Can we add firmer typings for the input?

File: modules/edit-modes/src/utils.ts [hunk 2]
```diff
@@ -457,3 +457,22 @@ function getEditHandlesForCoordinates(
   }
   return editHandles;
 }
+
+export function isNumeric(val: any) {
```
Comment (ibgreen) [inline modules/edit-modes/src/utils.ts:461]:
```suggestion
export function isNumeric(val: unknown): boolean {
```

Comment (ibgreen) [inline modules/edit-modes/src/utils.ts:461]: Add a JSDoc line for each function with a one liner explanation

**Review by ibgreen — APPROVED** [review]
Adding a 'screenSpace' parameter seems like a useful thing but why only for the translate mode? 

- Should we at least mention it in the docs? https://nebula.gl/docs/api-reference/modes/overview
- Without a test this will likely be fragile

**Comment by belopot** [comment]
@ibgreen  @igorDykhta I had this issue with the latest version (1.0.4). Do you have any advice?

**Comment by igorDykhta** [comment]
> @ibgreen @igorDykhta I had this issue with the latest version (1.0.4). Do you have any advice? https://github.com/uber/nebula.gl/assets/46432435/ddb4e0c4-bcde-491d-ac1c-7886f5019793

Have you tried to pass `screenSpace: true` for translate-mode?

**Comment by belopot** [comment]
@igorDykhta Yes I tried to pass `translateInScreenSpace: true` for `TranslateMode`. However it didn't work
image

**Comment by igorDykhta** [comment]
```
    modeConfig: {
      viewport,
      screenSpace: true,
    },
```

https://github.com/uber/nebula.gl/blob/b316c6348f2ddfdf819dd6128438b39c39fa6db7/modules/edit-modes/test/lib/translate-mode.test.ts#L59

**Pull request merged**